coffeebot.py

```diff
@@ -1,317 +1,306 @@
+                break
+
+        if self.trigger_check(slackclient, user, channel, clean):
+            message_processed = True
+
+        if not message_processed:
+            # We were mentioned, but we don't know what to do... Say
+            # something back to them.
+            channel.send_message('I am sorry {}, I can\'t do that.'.format(
+                self.mention(user)))
+
+    def handle_message(self, slackclient, event):
+        logger = logging.getLogger('handle_message')
+        logger.debug('message event: %s', event)
+        channel = slackclient.server.channels.find(event['channel'])
+
+        # If the user edits their message, we treat it as if it were a new
+        # message from that person.
+        if 'subtype' in event and event['subtype'] == 'message_changed':
+            event = event['message']
+        if 'user' not in event:
+            # Ignore events from non-users (i.e. coffebot app messages)
+            return
+        user = slackclient.server.users.find(event['user'])
+        text = event['text']
+
+        mentions = self.MENTION_RE.findall(text)
+        logger.info('Mentions: %s', mentions)
+
+        # Behaviours:
+        if self.USER_ID in mentions:
+            self.handle_mention_message(slackclient, user, channel, text)
+        elif ':coffee:' in text:
+            msg = 'Mmmm... :coffee:' + ':coffee:' * random.randint(0, 7)
+            channel.send_message(msg)
+
+    def loop(self, client):
+        logger = logging.getLogger('loop')
+        res = client.rtm_connect()
+        logger.debug('Connection result: %r', res)
+        if not res:
+            logger.error('Connection Failed.')
+            return
+
+        logger.info('Users: %s', client.server.users)
+        logger.info('Channels: %s', client.server.channels)
+        while True:
+            for event in client.rtm_read():
+                logger.debug('Event: %s', event)
+                if 'type' in event:
+                    # Call all handlers for the given event type.
+                    for handler in self.DISPATCH.get(event['type'], []):
+                        handler(client, event)
+            time.sleep(0.1)
 
 
 def main():
-  threads = []
-  for slack_workspace in models.SlackTeamAccessToken.query.filter(
-      models.SlackTeamAccessToken.coffee_bot_slack_access_token != None,
-  ):
-    sb = WrappedSlackBot(
-        slack_workspace.coffee_bot_slack_access_token,
-        slack_workspace.coffee_bot_slack_user_id,
-        slack_workspace.team_id,
-    )
-    threads.append(
-        threading.Thread(
-          target=sb.loop, args=(sb.client,)))
-  # Start all threads
-  for thread in threads:
-    thread.start()
-  # Wait for all threads to finish.
-  for thread in threads:
-    thread.join()
+    threads = []
+    for slack_workspace in models.SlackTeamAccessToken.query.filter(
+            models.SlackTeamAccessToken.coffee_bot_slack_access_token is not None,
```

Comment:
Please revert this, I actually want '!= None' (Which sqlalchemy will convert to "IS NOT NULL" in sql).

Reply:
Fixed in 224d270.